File: JSBASIC/191_min_diff.js
// 191	191_min_diff.js	Find the minimum difference between any two elements in an array.

let arr = [1, 5, 3, 19, 18, 25];
arr.sort((a, b) => a - b);
let minDiff = Infinity;
for (let i = 0; i < arr.length - 1; i++) {
    let diff = arr[i + 1] - arr[i];
    if (diff < minDiff) minDiff = diff;
}
console.log("Array:", arr);
console.log("Minimum difference:", minDiff);

File: JavaBasic/191_min_diff.java
// 191	191_min_diff.java	Find the minimum difference between any two elements in an array.
import java.util.Arrays;
public class _191_min_diff {
    public static void main(String[] args) {
        int[] arr = {1,5,3,19,18,25};
        Arrays.sort(arr);
        int minDiff = Integer.MAX_VALUE;
        for (int i=0;i<arr.length-1;i++) minDiff=Math.min(minDiff,arr[i+1]-arr[i]);
        System.out.println("Min difference: " + minDiff);
    }
}
